packages/core-p2p/src/peer.ts

```diff
@@ -210,7 +212,8 @@ export class Peer implements P2P.IPeer {
                 throw new PeerPingTimeoutError(delay);
             }
 
-            if (!(await peerVerifier.checkState(body, deadline))) {
+            this.verification = await peerVerifier.checkState(body, deadline);
+            if (!this.verification || this.verification.forked) {
```

Comment:
We shouldn't throw from here if the peer is on a legit chain, even if different from ours (fork). Throwing means "invalid state" or "could not verify the peer's state (thus assume it is invalid)".

Reply:
Changed in 245502691